src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -894,7 +916,14 @@ public void Dispose()
                 return;
             }
 
-            this.Disconnect("Peer disposed");
+            try
+            {
+                this.Disconnect("Peer disposed");
+            }
+            catch (Exception ex)
```

Comment:
Hiding a general exception is very bad this should throw if there is a bug or catch specifically what is the bug and then act on that.

Reply:
This is similar to line 952. The action taken is to display the error and continue with the rest of the `Dispose` calls.

Reply:
>Similar to this pattern followed elsewhere in this method:

This is a very specific reference, most of the times one should not ignore exceptions